RFC-0011-plugin-based-sql-invoked-functions-spi.md

```diff
@@ -0,0 +1,250 @@
+- One for native-sidecar deployments that includes all inlined SQL invoked functions currently registered under `BuiltInTypeAndFunctionNamespaceManager` but excludes functions overridden natively.
+
+This design aims to prevent conflicts in function resolution, especially when native functions are integrated into the coordinator in a sidecar-enabled cluster.
+
+
+## Background
+
+In sidecar-enabled clusters, users can configure the default function namespace that will be used to serve functions. The C++ functions are then loaded via an endpoint under this specified namespace.
+Previously, inlined SQL invoked functions were registered under the `presto.default` namespace (the default Java built-in namespace). Native functions pulled from the c++ sidecar were registered under a separate namespace like `native.default`.
+These native functions contain overridden optimized implementations of some of the inlined SQL invoked functions that the users would like to take advantage of. 
+However, this setup led to conflict resolution complexity. Presto would need to prioritize between two namespaces `presto.default` and `native.default` when resolving functions and such namespace-based prioritization is opaque and not ideal.
+
+To address this, the proposed change moves all inlined SQL invoked functions into a plugin-based namespace manager `BuiltInPluginFunctionNamespaceManager` which serves whatever the current default function namespace is. 
+This ensures:
+- All inlined SQL invoked functions are grouped together cleanly and loaded via a plugin.
+- Overridden native implementations of inlined SQL invoked functions are excluded from plugin registration.
+- The default serving catalog resolves functions seamlessly since all SQL invoked functions still live under the same default catalog but are now plugin loaded and managed by a plugin-based namespace manager `BuiltInPluginFunctionNamespaceManager`.
+
+### Goals
+- Introduce a new SPI method, `getSqlInvokedFunctionsSPI` for plugin-based registration of SQL invoked functions.
+- Create and wire a new plugin-based namespace manager `BuiltInPluginFunctionNamespaceManager` to manage these functions.
+- Enable native sidecar environments to load only the relevant subset of inlined SQL-invoked functions (i.e., excluding ones overridden natively).
+- Guarantee that inlined SQL invoked functions still work under the default catalog using the new plugin namespace manager.
+
+### Non-goals
+- This RFC does not Change the execution model for inlined SQL invoked functions - they are still evaluated in Java as earlier.
+- It does not modify function resolution rules outside of registration-time separation (i.e., no new prioritization logic is added).
+- It does not impact native (functions pulled from sidecar) or Java built-in function registration, which remains handled by the `NativeFunctionNamespaceManager` and `BuiltInTypeAndFunctionNamespaceManager` respectively.
```

Comment:
@kevintang2022  has a proposal which use this design to support native functions in Prestissimo https://docs.google.com/document/d/13tv1NYKWfy3v54kV1fyXFOYMHpCUkcEd6w8XSTyd2d0/edit?usp=sharing
which needs some extensions for this design.
I wonder if we can have one more built-in function plugin for native functions, so that native functions will also be under presto.default namespace, which will lead to seamless support for evaluation of functions of Java/Cpp/SQL at the same time.